cylc/flow/hostuserutil.py

```diff
@@ -99,13 +100,11 @@ def get_local_ip_address(target):
         """
 
         ipaddr = ""
-        try:
-            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-            sock.connect((target, 8000))
-            ipaddr = sock.getsockname()[0]
-            sock.close()
-        except IOError:
-            pass
+        with suppress(IOError):
+            with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as sock:
+                sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
```

Comment:
I don't understand why you need this line. Surely the line above defines `sock` in exactly the same way? 
The rest of this is pretty elegant.

Reply:
Oooops. Good spot @wxtim ! My mistake. Fixing it.